src/Interpreters/join_common.cpp

```diff
@@ -30,12 +29,42 @@ namespace ErrorCodes
 namespace
 {
 
+void insertFromNullableOrDefault(MutableColumnPtr & dst, const ColumnNullable * nullable_col)
```

Comment:
Can you explain why do we need to insert defaults in case of Null? Why don't just copy all data from nested column?

Reply:
Because it can contain some unexpected values. It can be reproduced with this example added to test https://github.com/ClickHouse/ClickHouse/blob/943b71e4ff61d0a4da58f7feb89df30187232bc8/tests/queries/0_stateless/00445_join_nullable_keys.sql#L41 Without setting default values, nested column will contain non-zero values for NULLs.